components/org.wso2.carbon.identity.api.user.association/org.wso2.carbon.identity.api.user.association.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/association/v1/core/UserAssociationService.java

```diff
@@ -46,22 +49,36 @@
     public List<UserDTO> getAssociationsOfUser(String userId) {
 
         try {
-            UserAccountAssociationDTO[] accountAssociationsOfUser = UserAccountConnectorServiceHolder
+            UserAccountAssociationDTO[] accountAssociationsOfUser = UserAccountAssociationServiceHolder
                     .getUserAccountConnector().getAccountAssociationsOfUser(userId);
             return getUserAssociationsDTO(accountAssociationsOfUser);
         } catch (UserAccountAssociationException e) {
             throw handleUserAccountAssociationException(e, "Error while getting associations of user: " + userId);
         }
     }
 
+    public List<FederatedUserDTO> getFederatedAssociationsOfUser(String userId) {
+
+        try {
+            AssociatedAccountDTO[] federatedAccountAssociationsOfUser = UserAccountAssociationServiceHolder
```

Comment:
+Add API

DELETE federated associations for a given user 
DELETE federated associations for the authenticated user

Reply:
Addressed.